Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,493 @@
+                    }
+
+                    IEnumerable<VariableExpressionAst> varsInLocalAssignments = FindVarsInAssignmentAsts(scriptBlockExpressionAst);
+                    if (varsInLocalAssignments != null)
+                    {
+                        AddAssignedVarsToSession(sessionName, varsInLocalAssignments);
+                    }
+
+                    GenerateDiagnosticRecords(
+                        FindNonAssignedNonUsingVarAsts(
+                            scriptBlockExpressionAst, 
+                            GetAssignedVarsInSession(sessionName)));
+
+                    return AstVisitAction.SkipChildren;
+                }
+                
+                return AstVisitAction.Continue;
+            }
+
+            /// <summary>
+            /// FindVarsInAssignmentAsts: Retrieves all assigned variables from an Ast:
+            /// Example: `$foo = "foo"` ==> the VariableExpressionAst for $foo is returned
+            /// </summary>
+            /// <param name="ast"></param>
+            private static IEnumerable<VariableExpressionAst> FindVarsInAssignmentAsts(Ast ast)
+            {
+                // Find all variables that are assigned within this ast
+                foreach (Ast foundAst in ast.FindAll(VarsInAssignments, true))
+                {
+                    var assignment = foundAst as AssignmentStatementAst;
+                    
+                    if (assignment.Left is VariableExpressionAst variable)
+                    {
+                        yield return variable;
+                    }
+                    else if (assignment.Left is ConvertExpressionAst conversion)
+                    {
+                        yield return conversion.Child as VariableExpressionAst;
+                    }
+                };
+            }
+
+            /// <summary>
+            /// VarsInAssignments: helper function to prevent allocation of closures for FindAll predicate.
+            /// </summary>
+            /// <param name="ast"></param>
+            private static bool VarsInAssignments(Ast ast)
+            {
+                return ast is AssignmentStatementAst;
+            }
+
+            /// <summary>
+            /// FindNonAssignedNonUsingVarAsts: Retrieve variables that are:
+            /// - not assigned before
+            /// - not prefixed with the 'Using' scope modifier
+            /// - not a PowerShell special variable
+            /// </summary>
+            /// <param name="ast"></param>
+            /// <param name="varsInAssignments"></param>
+            private static IEnumerable<VariableExpressionAst> FindNonAssignedNonUsingVarAsts(
+                Ast ast, IEnumerable<VariableExpressionAst> varsInAssignments)
+            {
+                // Find all variables that are not locally assigned, and don't have $using: scope modifier
+                foreach (Ast varAst in ast.FindAll(NonUsingNonSpecialVariables, true))

```

Comment:
The foreach will do the conversion for you, so you don't need to specify `Ast` here and have `as VariableExpressionAst` below. Just put `VariableExpressionAst` here